src/command/http-command.ts

```diff
@@ -93,18 +93,18 @@ const getInitialResult = () => ({
 	timings: {},
 });
 
-const allowedHttpProtocols = [ 'http', 'https', 'http2' ];
-const allowedHttpMethods = [ 'get', 'head' ];
+const allowedHttpProtocols = [ 'HTTP', 'HTTPS', 'HTTP2' ];
+const allowedHttpMethods = [ 'GET', 'HEAD' ];
 
 export const httpOptionsSchema = Joi.object<HttpOptions>({
-	type: Joi.string().valid('http').insensitive().required(),
```

Comment:
The fields are publicly case-insensitive. In theory, the are normalized to uppercase by the API, but it still seems safer to keep them insensitive here as well? Joi will correctly normalize them to uppercase if the uppercased versions are passed to `valid()`

Reply:
Was it actually wrong? Joi normalizes the value, so until now, it was converting everything to lowercase thanks to the `.insensitive()` part - that's why I was able to change what API does without breaking the probe communication.

Reply:
You are right, seems it is converting correctly. Ok, lets keep the `insensitive` part.